### ISSUE TITLE
ImplicitlyExpandDesignTimeFacades must trigger off of netstandard.dll once it is inbox

### ISSUE THREAD

**Issue by ericstj** [Microsoft/msbuild#2199]
Today ImplicitlyExpandDesignTimeFacades conditions the injection of facades based only on a reference to System.Runtime.

That works for contract-based PCLs, but not for netstandard libraries.  We'll want to also inject facades when netstandard.dll is referenced once netstandard.dll goes inbox.

/cc @alexghi

**Comment by rainersigwald** [comment]
* Approximately when is "once it is inbox" -- that is, do we need this for 15.3?
* This would just be roughly adding a check to the [is-any-dependency `System.Runtime` clause](https://github.com/Microsoft/msbuild/blob/54f7f93393cd96f237b419854ee8f2d7ab3acd71/src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs#L2239), right?

Possibly related: https://github.com/Microsoft/msbuild/pull/1520#issuecomment-271342350 -- do we need `ImplicitlyExpandDesignTimeFacades` at all?

**Comment by ericstj** [comment]
/cc @alexghi

We're having a meeting tomorrow to discuss this work where hopefully we can determine schedule.

Yes, I suspect that we could reuse the clause you point out.

>  do we need ImplicitlyExpandDesignTimeFacades at all?

I think that one is a bit independent and a riskier change.  

Certainly if ImplicitlyExpandDesignTimeFacades were removed this issue would not be required, but removing it and promoting all ReferenceDependencyPaths could either break existing apps if there happened to be type conflicts in the ReferenceDependencyPaths (not in the facades, but in other indirect dependencies).  Potentially if we filtered ReferenceDependencyPaths down to facades we could mitigate that risk.  I'd lean towards first doing this netstandard change and investigating the removal of ImplicitlyExpandDesignTimeFacades as a separate issue.
